service/pkg/db/db.go

```diff
@@ -185,7 +178,21 @@ func (c Config) buildConfig() (*pgxpool.Config, error) {
 		c.Database,
 		c.SSLMode,
 	)
-	return pgxpool.ParseConfig(u)
+	parsed, err := pgxpool.ParseConfig(u)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse pgx config: %w", err)
+	}
+	// Configure the search_path schema immediately on connection opening
+	parsed.AfterConnect = func(ctx context.Context, conn *pgx.Conn) error {
+		_, err := conn.Exec(ctx, fmt.Sprintf("SET search_path TO %s", c.Schema))
+		if err != nil {
+			slog.Error("failed to set database client search_path", slog.String("schema", c.Schema), slog.String("error", err.Error()))
+			return err
+		}
+		slog.Info("successfully set database client search_path", slog.String("schema", c.Schema))
```

Comment:
Should we set this to debug?

Reply:
Now that we can reliably reproduce and verify it's not an issue, I'm good with that. Good call. Will fix now.